core/modules/replica/HttpQservMonitorModule.cc

```diff
@@ -0,0 +1,477 @@
+            ::parseFieldIntoJson<string>( __func__, row, "qTemplate",      resultRow);
+            ::parseFieldIntoJson<string>( __func__, row, "qMerge",         resultRow, "");
+            ::parseFieldIntoJson<string>( __func__, row, "status",         resultRow);
+            ::parseFieldIntoJson<string>( __func__, row, "submitted",      resultRow);
+            ::parseFieldIntoJson<long>(   __func__, row, "submitted_sec",  resultRow);
+            ::parseFieldIntoJson<string>( __func__, row, "completed",      resultRow, "");
+            ::parseFieldIntoJson<long>(   __func__, row, "completed_sec",  resultRow, 0);
+            ::parseFieldIntoJson<string>( __func__, row, "returned",       resultRow, "");
+            ::parseFieldIntoJson<long>(   __func__, row, "returned_sec",   resultRow, 0);
+            ::parseFieldIntoJson<string>( __func__, row, "messageTable",   resultRow, "");
+            ::parseFieldIntoJson<string>( __func__, row, "resultLocation", resultRow, "");
+            ::parseFieldIntoJson<string>( __func__, row, "resultQuery",    resultRow, "");
+            result["queries_past"].push_back(resultRow);
+        }
+    }
+    sendData(resp, result);
+}
+
+
+void HttpQservMonitorModule::_userQuery(qhttp::Request::Ptr const& req,
+                                        qhttp::Response::Ptr const& resp) {
+    debug(__func__);
+
+    auto const id = stoull(req->params.at("id"));
+
+    debug(__func__, " id=" + to_string(id));
+
+    json result;
+    sendData(resp, result);
+}
+
+
+json HttpQservMonitorModule::_getQueries(json& workerInfo) const {
+
+    // Find identifiers of all queries in the wait queues of all schedulers
+    set<QueryId> qids;
+    for (auto&& scheduler: workerInfo.at("processor").at("queries").at("blend_scheduler").at("schedulers")) {
+        for (auto&& entry: scheduler.at("query_id_to_count")) {
+            qids.insert(entry[0].get<QueryId>());
+        }
+    }
+
+    // Connect to the master database
+    // Manage the new connection via the RAII-style handler to ensure the transaction
+    // is automatically rolled-back in case of exceptions.
+
+    auto const config = controller()->serviceProvider()->config();
+    database::mysql::ConnectionHandler const h(
+        database::mysql::Connection::open(
+            database::mysql::ConnectionParams(
+                config->qservMasterDatabaseHost(),
+                config->qservMasterDatabasePort(),
+                "root",
+                Configuration::qservMasterDatabasePassword(),
+                "qservMeta"
+            )
+        )
+    );
+
+    // Extract descriptions of those queries from qservMeta
+
+    json result;
+
+    if (not qids.empty()) {
+        h.conn->execute([&](decltype(h.conn) conn) {
+            conn->begin();
+            conn->execute(
+                "SELECT * FROM " + conn->sqlId("QInfo") + "  WHERE " + conn->sqlIn("queryId", qids)
+            );
```

Comment:
I think it's better to use QMeta API for qmeta access, not hand-written queries.

Reply:
No way 😹 I don't want to break my beautifully crafter Replication system by relying on an API which won't even do reconnect if the MySQL service goes down. I may come back to this idea after proper improvements will be done to that API. I'm 100% sure this will bite us really hard after migrating Qserv to Kubernetes.

Reply:
QMeta uses sql/MysqlConnection which does try to re-connect is connection is lost.

_beautifully crafter_? I'm not going to comment on this but you know what I can say 😉